test/hotspot/jtreg/runtime/8176717/TestInheritFD.java

```diff
@@ -291,7 +305,12 @@ public static void main(String[] args) throws InterruptedException {
                     Thread.sleep(300 * 1000);
                 }
             } catch (Exception e) {
-                System.out.println("Exception was thrown: " + e.toString());
+                if (e instanceof TimeoutException) {
+                    TimeoutException te = (TimeoutException)e;
+                    System.out.println("(Third VM) Timed out waiting for second VM: " + te.toString());
+                } else {
+                    System.out.println("(Third VM) Exception was thrown: " + e.toString());
+                }
                 throw e;
```

Comment:
I do not know if this gives us any extra information from the single `println`. `e.toString()` ought to print the exception type, correct? If you really think it is worth having two different print statements, I think it would be better having one catch statement for each exception (both rethrowing), I guess the compiler would then complain if one exception source was removed. Either way, the casting to the TimeoutException is not necessary.

Reply:
I need to look at it a bit more, I think this is not correct.

Reply:
The original code seems to try to catch one of two runtime exceptions:  CancellationException or CompletionException. The code you added can not be correct as a TimeoutException is a checked exception (and not thrown if I understand the code correctly).

Reply:
I agree with Leo. IIUC if we timeout then a `CompletionException` will be thrown from `join()` with `TimeoutException` as the cause.

Reply:
You're correct, I didn't understand the original code (I thought `TimeoutException` was thrown). I think the added information in the printing message is worth it, I just had to figure out why a `TimeoutException` was thrown and the extra context helps.